lighthouse-core/test/scripts/i18n/collection-util-test.js

```diff
@@ -0,0 +1,384 @@
+    };`;
+
+    const ast = esprima.parse(justUIStrings, {comment: true, range: true});
+
+    const stmt = ast.body[0];
+    const prop = stmt.declarations[0].init.properties[0];
+    const res = collect.computeDescription(ast, prop, 'Hello World', 0);
+    expect(res.description).toBe(undefined);
+  });
+
+  it('collects complex description', () => {
+    const justUIStrings =
+    `const UIStrings = {
+      /** 
+       * @description Tagged description for Hello World.
+       */
+      message: 'Hello World',
+    };`;
+
+    const ast = esprima.parse(justUIStrings, {comment: true, range: true});
+
+    const stmt = ast.body[0];
+    const prop = stmt.declarations[0].init.properties[0];
+    const res = collect.computeDescription(ast, prop, 'Hello World', 0);
+    expect(res.description).toBe('Tagged description for Hello World.');
+  });
+
+  it('collects complex description with example', () => {
+    const justUIStrings =
+    `const UIStrings = {
+      /** 
+       * @description Tagged description for Hello World.
+       * @example {variable} Variable example.
+       */
+      message: 'Hello World {variable}',
+    };`;
+
+    const ast = esprima.parse(justUIStrings, {comment: true, range: true});
+
+    const stmt = ast.body[0];
+    const prop = stmt.declarations[0].init.properties[0];
+    const res = collect.computeDescription(ast, prop, 'Hello World {variable}', 0);
+    expect(res.description).toBe('Tagged description for Hello World.');
+    expect(res.examples['variable']).toBe('Variable example.');
+  });
+
+  it('errors when example given without variable', () => {
+    const justUIStrings =
+    `const UIStrings = {
+      /** 
+       * @description Tagged description for Hello World.
+       * @example {variable} Variable example.
+       */
+      message: 'Hello World',
+    };`;
+
+    const ast = esprima.parse(justUIStrings, {comment: true, range: true});
+
+    const stmt = ast.body[0];
+    const prop = stmt.declarations[0].init.properties[0];
+    expect(() => collect.computeDescription(ast, prop, 'Hello World', 0))
+      .toThrow(/Example missing ICU replacement/);
+  });
+
+  it('errors when variable has no example', () => {
+    const justUIStrings =
+    `const UIStrings = {
+      /** 
+       * @description Tagged description for Hello World.
+       */
+      message: 'Hello World {variable}',
+    };`;
+
+    const ast = esprima.parse(justUIStrings, {comment: true, range: true});
+
+    const stmt = ast.body[0];
+    const prop = stmt.declarations[0].init.properties[0];
+    expect(() => collect.computeDescription(ast, prop, 'Hello World {variable}', 0))
+      .toThrow(/Variable 'variable' is missing example comment/);
+  });
+});
+
+describe('Convert Message to Placeholder', () => {
+  it('passthroughs a basic message unchanged', () => {
+    const message = 'Hello World.';
+    const res = collect.convertMessageToPlaceholders(message, undefined);
+    expect(res.message).toBe(message);
```

Comment:
could we assert on the whole object? would be super helpful for looking at what everything returns shape-wise